src/toxcore/dht/codec.rs

```diff
@@ -52,14 +55,47 @@ pub enum EncodeError {
 }
 
 /// Struct to use for {de-,}serializing DHT UDP packets.
-#[derive(Clone, Debug, Eq, PartialEq)]
-pub struct DhtCodec;
+#[derive(Clone, Debug)]
+pub struct DhtCodec {
+    /// Incoming packets count for Udp
+    incoming: Arc<RwLock<u64>>,
```

Comment:
RwLock -> atomic

Reply:
What should happen on overflow?

Reply:
That's a good question. I don't know. Please calculate an estimate time to get an overflow.

Reply:
I asked because it will be worse if we use AtomicUsize here and build node for 32 bit architecture. Though for node it's unlikely to happen but it seems can happen for long running client.